### PR TITLE
test: add parameter to task to allow fetching things from origin

### PR DESCRIPTION
**Pull request by spring1843**
Fixes # 

**Description**

Will allow us to do things like running tests on PRs

**How was this change tested?**

* Local copy

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates
- [ ] Yes, issue opened: # 
- [x] No

**Release Note**

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/suites/common/run-test.yaml [hunk 1]
```diff
@@ -25,8 +29,16 @@ spec:
     workingDir: $(workspaces.ws.path)
     script: |
       echo "Cloning $(params.git-repo-url)"
+      echo "Fetching - Optional: $(params.git-fetch-origin)"
       echo "Checking out git ref: $(params.git-ref)"
-      git clone $(params.git-repo-url).git .
+
+      git clone $(params.git-repo-url).git
+
+      if [ "$(params.git-fetch-origin)" != "" ]; then
```
Comment (bwagner5) [inline test/suites/common/run-test.yaml:37]: I don't think we need this parameter. Why not just fetch all the time and build the fetch command from the git-ref param?  

You could also just expect the `pull/ID/head:#1234` to be the input to git-ref if you want a PR and then `git fetch ${params.git-ref}` all the time.

File: test/suites/common/run-test.yaml [hunk 2]
```diff
@@ -26,8 +26,11 @@ spec:
     script: |
       echo "Cloning $(params.git-repo-url)"
       echo "Checking out git ref: $(params.git-ref)"
-      git clone $(params.git-repo-url).git .
-      git checkout $(params.git-ref)
+
+      git clone $(params.git-repo-url).git
+
+      git fetch origin ${params.git-ref}
+      git checkout $(echo ${params.git-ref} | cut -d":" -f2)
```
Comment (bwagner5) [inline test/suites/common/run-test.yaml:33]:
```suggestion
      git fetch origin $(params.git-ref)
      git checkout $(echo $(params.git-ref) | cut -d":" -f2)
```

File: test/suites/common/run-test.yaml (hunk 1 above)
Comment (spring1843) [reply test/suites/common/run-test.yaml]: done good idea

**Review by bwagner5 — APPROVED** [review]
LGTM

**Pull request merged**